### PR TITLE
feat: add graphql operation to fetch and cache issues

### PR DESCRIPTION
**Pull request by Extheoisah**
This PR does the following: 
- [x] modifies the tailwind config to extend the base colors instead of directly modifying them.
- [x] caches issues using Apollo's type policies
- [ ] adds cursor pagination to fetch other issues if present

### CODE REVIEW

File: graphql/queries/get-issues.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+                        title
+                        url
+                        number
+                        publishedAt
+                    }
+                }
+            }
+        }
+    }
+`
```
Comment (BalogunofAfrica) [inline graphql/queries/get-issues.ts:23]:
```suggestion
`

export function constructRepoQueries(
    inputs: Array<{
        owner: string
        name: string
        labels: string[]
        states: string[]
    }>
) {
    const queryParts = inputs.map((input, index) => {
        const labelsFormatted = JSON.stringify(input.labels)
        const statesFormatted = input.states.map((state) => state).join(", ")

        return `
      repo${index}: repository(owner: "${input.owner}", name: "${input.name}") {
        issues(first: 10, labels: ${labelsFormatted}, states: ${statesFormatted}) {
          edges {
            node {
              title
              url
              number
              publishedAt
            }
          }
        }
      }
    `
    })

    return gql`
         query GetIssues {
            ${queryParts.join("\n")}
         }
    `
}
```

File: hooks/useGetRepositoryIssues.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    })
+
+    return {
+        issues: data?.repository.issues.edges.map(
+            (edge: { node: Issue }) => edge.node
+        ) as Issue[],
+        loading,
+        error
+    }
+}
```
Comment (BalogunofAfrica) [inline hooks/useGetRepositoryIssues.tsx:34]:
```suggestion
import type { Issue, Project } from "@/types";
import { useQuery } from "@apollo/client";

import { constructRepoQueries } from "../graphql/queries/get-issues";
import projectJson from "../public/opensource-projects/index.json";

const projectsMetadata = Object.entries(projectJson as Project).map(
	([_, p]) => ({
		name: p.name,
		owner: p.org,
		languages: p.lang,
	}),
);
const labels = ["good first issue", "bug", "help wanted"];
const states = ["OPEN"];
const reposWithLabels = projectsMetadata.map((n) => ({
	...n,
	labels,
	states,
}));

export function useGetRepositoryIssues() {
	const { data, loading, error } = useQuery(
		constructRepoQueries(reposWithLabels),
		{
			fetchPolicy: "cache-and-network",
			nextFetchPolicy: "cache-first",
		},
	);

	const issues = data
		? Object.entries(data).flatMap(([_, issuesData], index) => {
				if (!issuesData) return [];
				return ((issuesData as any)?.issues?.edges as any[]).map(
					(edge: { node: Issue }) => ({
						...edge.node,
						owner: projectsMetadata[index].owner,
						languages: projectsMetadata[index].languages,
						repo: projectsMetadata[index].name,
					}),
				);
			})
		: [];

	return {
		issues,
		loading,
		error,
	};
}
```

File: components/repository-issues.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+const RepositoryIssues = ({
+    owner,
+    name,
+    languages
+}: {
+    owner: string
+    name: string
+    languages: string[]
+}) => {
```
Comment (BalogunofAfrica) [inline components/repository-issues.tsx:20]:
```suggestion
const RepositoryIssues = () => {
```

File: components/repository-issues.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                            {issue.title}
+                        </h3>
+                    </div>
+                    <div className="flex flex-wrap mt-2">
+                        {languages.map((lang) => (
+                            <Badge key={lang} name={lang} className="mr-2" />
+                        ))}
+                    </div>
+                </Link>
+            ))}
```
Comment (BalogunofAfrica) [inline components/repository-issues.tsx:72]:
```suggestion
{issues.map((issue) => (
				<Link
					key={issue.number + issue.publishedAt}
					className="flex flex-col justify-between min-h-[180px] md:min-h-36 rounded-lg border border-gray-200 bg-white p-4 shadow-sm transition-all hover:scale-105 hover:shadow-md"
					href={issue.url}
					rel="noopener noreferrer"
					target="_blank"
				>
					<div className="flex flex-col justify-between">
						<div className="flex items-center justify-between">
							<span className="text-sm font-medium text-gray-500">
								{`${issue.repo}/${issue.owner}`}
							</span>
							<span className="text-xs text-gray-500">
								{new Date(issue.publishedAt).toLocaleDateString("en-US", {
									year: "numeric",
									month: "short",
									day: "numeric",
								})}
							</span>
						</div>
						<h3 className="mt-2 text-lg md:text-base font-medium text-gray-900">
							<span className="text-gray-500">#{issue.number}</span>{" "}
							{issue.title}
						</h3>
					</div>
					<div className="flex flex-wrap mt-2">
						{issue.languages.map((lang) => (
							<Badge key={lang} name={lang} className="mr-2" />
						))}
					</div>
				</Link>
			))}
```

File: app/contribute/page.tsx [hunk 5]
```diff
@@ -10,27 +15,41 @@ const Contribute = async () => {
...
+                            {Object.entries(Projects as Project).map(
+                                ([key, project]) => (
+                                    <RepositoryIssues
+                                        key={key}
+                                        owner={project.org}
+                                        name={project.name}
+                                        languages={project.lang}
+                                    />
+                                )
+                            )}
```
Comment (BalogunofAfrica) [inline app/contribute/page.tsx:48]:
```suggestion
             <RepositoryIssues />
```

File: components/repository-issues.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import React from "react"
+
+import { useGetRepositoryIssues } from "@/hooks/useGetRepositoryIssues"
+
+import Badge from "./badge"
+import Skeleton from "./skeleton"
+
+const RepositoryIssues = () => {
+    const { error, issues, loading } = useGetRepositoryIssues()
+    console.log({ loading })
```
Comment (IgboPharaoh) [inline components/repository-issues.tsx:13]: We can remove this console.log

**Review by IgboPharaoh — COMMENTED** [review]
looks good to me just a suggestion to remove a nit

File: components/repository-issues.tsx (hunk 6 above)
Comment (Extheoisah) [reply components/repository-issues.tsx]: addressed -> https://github.com/bitcoin-dev-project/bitcoin-dev-project/pull/41/commits/31cfca3c7933387a8f8d2d6b1c0789d6dadb3ff6

**Pull request merged**